.github/workflows/deploy-to-production.yml

```diff
@@ -7,14 +7,22 @@ on:
       - 'Deploy to staging'
     types:
       - completed
+    branches:
+      - main
 
 permissions:
   contents: read
   id-token: write
 
 jobs:
   deploy:
-    if: ${{ github.event.workflow_run.conclusion == 'success' || github.event_name == 'workflow_dispatch' }}
+    if: ${{
```

Comment:
Does this block deploys to production using dispatch? I think if we want to know that a staging dispatch triggered the upstream workflow (e.g. the staging workflow) then we'll want to produce an output that can be consumed there that indicates the upstream was kicked off by dispatch?